include/windows/osd.h

```diff
@@ -102,7 +102,81 @@ do						\
 
 #define MIN min
 #define MAX max
-#define OFI_UNUSED(p) (void)(p)
+#define OFI_UNUSED UNREFERENCED_PARAMETER
+
+/* Atomic memory access functions */
+static inline
+long ofi_atomic_inc32(volatile long *ptr)
+{
+	return InterlockedIncrement(ptr);
+}
+
+static inline
+long long ofi_atomic_inc64(volatile long long *ptr)
+{
+	return InterlockedIncrement64(ptr);
+}
+
+static inline
+long ofi_atomic_dec32(volatile long *ptr)
+{
+	return InterlockedDecrement(ptr);
+}
+
+static inline
+long long ofi_atomic_dec64(volatile long long *ptr)
+{
+	return InterlockedDecrement64(ptr);
+}
+
+static inline
+long ofi_atomic_add32(volatile long *ptr, long value)
+{
+	return InterlockedAdd(ptr, value);
+}
+
+static inline
+long long ofi_atomic_add64(volatile long long *ptr, long long value)
+{
+	return InterlockedAdd64(ptr, value);
+}
+
+static inline
+long ofi_atomic_sub32(volatile long *ptr, long value)
+{
+	return InterlockedAdd(ptr, -value);
+}
+
+static inline
+long long ofi_atomic_sub64(volatile long long *ptr, long long value)
+{
+	return InterlockedAdd64(ptr, -value);
+}
+
+static inline
+void* ofi_val_compare_and_swap_ptr(volatile void **ptr, void *ptr1, void *ptr2)
+{
+	return InterlockedCompareExchangePointer((PVOID *)ptr, ptr1, ptr2);
+}
+
+static inline
+long ofi_val_compare_and_swap32(volatile long *ptr, long oldval, long newval)
+{
+	return InterlockedCompareExchange(ptr, oldval, newval);
+}
+
+static inline
+long long ofi_val_compare_and_swap64(volatile long long *ptr, long oldval, long newval)
+{
+	return InterlockedCompareExchange64(ptr, oldval, newval);
+}
+
+static inline
+unsigned char ofi_val_compare_and_swap128(volatile long long *ptr, long long oldval,
+					  long long newval, long long *compare)
+{
+	return InterlockedCompareExchange128(ptr, oldval, newval, compare);
+}
```

Comment:
Yes, I agree. When I'll move netdir_buf.h content as a part of common code, I should found the proper solution for that since this function is used.